### ISSUE TITLE
Torch UserWarning: None of the inputs have requires_grad=True

### ISSUE THREAD

**Issue by bmmo** [petko-nikolov/semseg.pytorch#5]
I get this warning with the FC-DenseNet implementation for inference:

> python3.7/site-packages/torch/utils/checkpoint.py:25: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
>  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")

This happens even when using `torch.no_grad()` and `model.eval()`
I know it's related to the efficient checkpointing but thought I would mention it nonetheless.

**Comment by petko-nikolov** [comment]
Thanks for reporting this! I will take a look at it soon.
